app/workers/streams/twitch_webhook_registration_worker.rb

```diff
@@ -1,6 +1,7 @@
 module Streams
-  class TwitchWebhookRegistrationJob < ApplicationJob
-    queue_as :twitch_webhook_registration
+  class TwitchWebhookRegistrationWorker
+    include Sidekiq::Worker
+    sidekiq_options queue: :default, retry: 10
 
     def perform(user_id, service = TwitchWebhook::Register)
```

Comment:
Can we remove this service argument to simplify the job since we are not using it? Plus if we wanted to use it we would need it to be a string and then constantize it.

Reply:
It is used in line 10:

```ruby
service.call(user)
```

Also, it shouldn't need to be a string and constantized. Classes are normal objects, they can just be passed around like everything else:

```ruby
class A ; end

def hello(klass = A)
  klass.new
end

hello
#=> #<A:0x00000001143080f8>
```

It also won't get serialized into a string by putting it on Redis, since we `perform_async` with only the user id, which means `service` will be assigned during job run time. I just tried it out locally with a `binding.pry` in the worker:

```ruby
# Enqueue
Streams::TwitchWebhookRegistrationWorker.perform_async(u.id)
#=> "d43b2b2b8f8f320532e2c58a"

# binding.pry in running job
[1] dev.to(#<Streams::TwitchWebhookRegistrationWorker>)> service.class
Class < Module
```

Reply:
Right, but having it as an optional argument means that we have the option to send a different value in when we enqueue the job. If we chose to send a different value in, that value will be a string and this code, even if the string is a valid class, will break. 

In addition, this worker is very specific, `TwitchWebhookRegistrationWorker` so I wouldn't expect us to want to send any other service besides `TwitchWebhook::Register` which means I think we may as well remove it as an argument.

Reply:
Got your point now. I was operating under the assumption that this argument is mostly there in case we want to inject a dummy during testing (when calling `perform`), but that’s not worth the hassle.

Reply:
I removed it now. Sorry for the misunderstanding, I was really under the impression the `service` argument was there for testing purposes only, for which it worked well enough.

Reply:
> I removed it now.

Did you push up that change yet? I'm still seeing the argument in the PR

Reply:
I thought I had, but apparently that wasn't true. Pushed now.

Reply:
Been there, gotten that badge